python/dgl/graphbolt/utils/sample_utils.py

```diff
@@ -267,3 +267,182 @@ def unique_and_compact_node_pairs(
         unique_nodes = list(unique_nodes.values())[0]
 
     return unique_nodes, compacted_node_pairs
+
+
+def duplicated_and_compact_node_pairs(
```

Comment:
the method body and `unique_and_compact_node_pairs` seems have much in common. we'd better avoid code duplication

Reply:
deduplicated? not_deduplicated?

Just name it to compact_node_pairs should be fine. Just compact no unique.